code/processes/metrics.q

```diff
@@ -35,49 +41,39 @@ metrics:{[syms]
    / add allday window
    if[.metrics.enableallday;
     if[not all syms in key .metrics.start;.metrics.start::exec first time by sym from sumstab];
-    t:`sym`timediff xasc t,select sym,timediff:0Nn,vwap:sumsps%sumssize,twap:sumspricetimediff%.z.p - .metrics.start[sym] from latest where sym in syms];
-
-   t
-  
+    t:`sym`timediff xasc t,select sym,timediff:0Nn,vwap:sumsps%sumssize,twap:sumspricetimediff%.z.p - .metrics.start[sym] from latest where sym in syms];t  
  }
 
 \d .metrics 
 
 / check for TP connection
 notpconnected:{[]
-	0 = count select from .sub.SUBSCRIPTIONS where proctype in ((),`tickerplant), active}
+	0 = count select from .sub.SUBSCRIPTIONS where proctype in tickerplanttypes,active}
 
 / get handle for TP & subscribe
 subscribe:{
-  / get handle
-  if[count s:.sub.getsubscriptionhandles[`tickerplant;();()!()];
-  subproc:first s;
-  / if got handle successfully, subsribe to trade table
-  .lg.o[`subscribe;"subscribing to ", string subproc`procname];
-  :.sub.subscribe[`trade;`;0b;0b;subproc]]};
+   / get handle
+   if[count s:.sub.getsubscriptionhandles[`tickerplant;();()!()];
```

Comment:
I think this is a another scenario where the condition can be flipped, so:
```
if[0=count s:.sub.getsubscriptionhandles[`tickerplant;();()!()];:()];
subproc:first s;
...
```
Essentially the same thing but neater IMO. Would need a comment just to say "exiting if no handles are found" or similar.